### DISCUSSION TITLE
Is there a way to get pactum to allow self-signed certificates?

### DISCUSSION THREAD

**Discussion by caveman-dick** [pactumjs/pactum#252] [Q&A]
I'm trying to test my api locally which has self-signed certs and I'm getting this error:

 console.error
    {
      "code": "DEPTH_ZERO_SELF_SIGNED_CERT"
    }

I've tried setting process.env.NODE_TLS_REJECT_UNAUTHORIZED=0 but that doesn't work. Is there a way I can allow self-signed certificates via pactum?

I'm using jest as the test runner

Thanks

**Comment by crystalfp** [comment]
I have the same problem and my self-signed certificate is generated with:
```
openssl req -x509 -newkey rsa:2048 -nodes -sha256 -subj '/CN=localhost' -days 365 -keyout localhost-privkey.pem -out localhost-cert.pem
```
With this certificate my client can connect to my server without problems.

**Reply by crystalfp** [reply]
Now I'm experimenting with [mkcert ](https://github.com/FiloSottile/mkcert) and the error changed. Now Pactun complains that it is: `UNABLE_TO_VERIFY_LEAF_SIGNATURE`.
The certificate for the server is generated this way:
```
./mkcert-v1.4.4-windows-amd64.exe \
-cert-file "./localhost-cert.pem" \
-key-file "./localhost-privkey.pem" \
localhost 127.0.0.1 ::1
```

**Reply by ASaiAnudeep** [reply]
pactum uses [phin](https://github.com/ethanent/phin) as its http library. It allows us to use the nodejs internal http options using [withCore](https://pactumjs.github.io/api/requests/withCore.html) method which might help.

**Reply by crystalfp** [reply]
Thanks for the suggestion! But I fear I misunderstand something fundamental regarding https.
